controller/app.go

```diff
@@ -55,6 +56,12 @@ func (r *AppRepo) Add(data interface{}) error {
 	}
 	meta := metaToHstore(app.Meta)
 	if err := r.db.QueryRow("INSERT INTO apps (app_id, name, meta, strategy) VALUES ($1, $2, $3, $4) RETURNING created_at, updated_at", app.ID, app.Name, meta, app.Strategy).Scan(&app.CreatedAt, &app.UpdatedAt); err != nil {
+		if e, ok := err.(*pq.Error); ok && e.Code.Name() == "unique_violation" && e.Constraint == "apps_name_idx" {
```

Comment:
This code is used in a few places, I think we should refactor it to something like:

```go
if postgres.IsUniquenessError(err, "apps_name_idx") {
	...
}
```